src/functional/iterator/embedded.py

```diff
@@ -701,25 +699,57 @@ def _get_axes(
     )
 
 
+def _single_vertical_idx(indices, column_axis_idx, column_index):
+    transformed = tuple(
+        index if i != column_axis_idx else column_index for i, index in enumerate(indices)
+    )
+    return transformed
+
+
 def _make_tuple(
     field_or_tuple: LocatedField | tuple,  # arbitrary nesting of tuples of LocatedField
     indices: FieldIndexOrIndices,
     *,
-    as_column: bool = False,
-) -> tuple | Column:  # arbitrary nesting of tuples of field values or `Column`s
+    column_axis: Optional[Tag] = None,
+) -> npt.DTypeLike | Column | tuple:  # arbitrary nesting of tuples of field values or `Column`s
     if isinstance(field_or_tuple, tuple):
-        return tuple(_make_tuple(f, indices, as_column=as_column) for f in field_or_tuple)
+        if column_axis is not None:
+            assert _column_range
+            # construct a Column of tuples
+            column_axis_idx = _axis_idx(_get_axes(field_or_tuple), column_axis)
+            first = tuple(
+                _make_tuple(f, _single_vertical_idx(indices, column_axis_idx, _column_range.start))
+                for f in field_or_tuple
+            )
+            col = Column(
+                _column_range.start, np.zeros(len(_column_range), dtype=_column_dtype(first))
+            )
+            col[0] = first
+            for i in _column_range[1:]:
+                col[i] = tuple(
+                    _make_tuple(f, _single_vertical_idx(indices, column_axis_idx, i))
+                    for f in field_or_tuple
+                )
+            return col
+        else:
+            return tuple(_make_tuple(f, indices) for f in field_or_tuple)
     else:
-        data = field_or_tuple[indices]
-        if as_column:
+        data = field_or_tuple.field_getitem(indices)
+        if column_axis is not None:
             # wraps a vertical slice of an input field into a `Column`
-            return Column(
-                0, data
-            )  # TODO(havogt) when we support LocatedFields with origin (i.e. non-zero origin), kstart needs to be adapated here
+            assert _column_range is not None
+            return Column(_column_range.start, data)
         else:
             return data
 
 
+def _axis_idx(axes: Sequence[common.Dimension], axis: Tag) -> int:
+    for i, a in enumerate(axes):
+        if a.value == axis:
+            return i
+    raise AssertionError()
```

Comment:
Even if it's an internal error that should never happen, I would add a descriptive message just for developers reading  the code.